docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+- After handling a transaction, Hedera Services will search within the next `autorenew.numberOfEntitiesToCheck` for upto `autorenew.maxNumberOfEntitiesToRenew` entities that expired then try to renew these entities.
+- After the grace period, if the `expirationTime` of an entity is not extended, it will be deleted from the system.
+
+## Implementation
+Hedera Services will perform a circular scanning of entities, meaning after we reach the last entity in the system, we will go back scanning from the first entity in the system.
+
+When trying to renew an entity:
+1. Calculate the fee to extend the entity's `expirationTime` for another `autoRenewPeriod`.
+2. If the `autoRenewAccount` of the entity has enough balance to cover this fee:
+  - extend the entity's `expirationTime` for another `autoRenewPeriod`.
+  - otherwise, translate the remaining balance of the `autoRenewAccount` into an extension, preferably proportional to the fee calculated in step 1, then extend accordingly.
+3. If the grace period also passes, delete the entity from the system.
+
+For restart and reconnect: The last scanned entity must be in the state for synchronization.
+
+## Autorenewal record
+After autorenewing an entity, Hedera Services will generate a `TransactionRecord` that serves as an autorenewal record and contains the following:
+
+| Field | Type | Label | Description |
+|---|---|---|---|
+| receipt | TransactionReceipt | | receipt will contain either an accountID, a fileID, a contractID, a topicID or a tokenID that got autorenewed |
+| transactionHash | bytes | | empty |
+| consensusTimestamp | Timestamp | | The consensus timestamp of the autorenewal |
+| transactionID | TransactionID | | { empty transactionValidStart, autoRenewAccount } |
+| memo | string | | "Entity {ID} was automatically renewed. New expiry: {newExpiry}" |
+| transactionFee | uint64 | | The fee charged for the autorenewal of the entity |
+| contractCallResult | ContractFunctionResult | | empty |
+| contractCreateResult | ContractFunctionResult | | empty |
+| transferList | TransferList | | {(autoRenewAccount, -transactionFee), (defaultFeeCollectionAccount, transactionFee)} |
+| tokenTransferLists | TokenTransferList | repeated | empty |
+| scheduleRef | ScheduleID | | empty |
+
+The main difference between an autorenewal record and a regular `TransactionRecord` is that it has an empty `transactionHash` and an empty `transactionID.transactionValidStart`. There were older versions of Hedera Services that did not have `transactionHash` in a `TransactionRecord`, but an empty `transactionID.transactionValidStart` will guarantee that the `TransactionRecord` was generated in place of an autorenewal record.
```

Comment:
Note without a valid `transactionID.transactionValidStart` it will be difficult to uniquely identify and record this state change from a viewers perspective as we can with all current transaction ids